src/txn/txn_rollback_to_stable.c

```diff
@@ -304,21 +304,17 @@ __rollback_check_if_txnid_non_committed(WT_SESSION_IMPL *session, uint64_t txnid
         return (false);
 
     /*
+     * The order here is important: anything newer than or equal to the maximum ID we saw when
+     * taking the snapshot should be invisible, even if the snapshot is empty.
+     *
      * Snapshot data:
+     *  ids >= recovery_ckpt_snap_max are non committed,
      *	ids < recovery_ckpt_snap_min are committed,
-     *	ids > recovery_ckpt_snap_max are non committed,
      *	everything else is committed unless it is found in the recovery_ckpt_snapshot array.
      */
-    if (txnid < conn->recovery_ckpt_snap_min)
-        return (false);
-    else if (txnid > conn->recovery_ckpt_snap_max)
+    if (txnid >= conn->recovery_ckpt_snap_max)
```

Comment:
Looking at the comment in `__txn_visible_id`, I realised another issue that the order to compare snap max and min also matters. I bring the code to align with the `__txn_visible_id` implementation.

Reply:
I think the order refers to this condition `conn->recovery_ckpt_snapshot_count == 0`.  As long as we compare snap max before checking that, it should be OK.